src/tests/ftest/io/mdtest_small.py

```diff
@@ -44,6 +44,62 @@ def test_mdtest_small(self):
             depth of hierarchical directory structure: 0|5
         :avocado: tags=all,pr,hw,mdtest,mdtestsmall
         """
-        mdtest_flags = self.params.get("flags", "/run/mdtest/*")
-        self.mdtest_cmd.flags.update(mdtest_flags)
-        self.execute_mdtest()
+        # local params
+        mdtest_flags = self.params.get("mdtest_flags", "/run/mdtest/*")
+        mdtest_api = self.params.get("mdtest_api", "/run/mdtest/*")
+        read_write = self.params.get("read_write", "/run/mdtest/*")
+        branch_factor = self.params.get("branch_factor", "/run/mdtest/*")
+        dir_depth = self.params.get("dir_depth", "/run/mdtest/*")
+        num_of_items = self.params.get("num_of_items", "/run/mdtest/*")
+        number_of_files_dirs = self.params.get("number_of_files_dirs",
+                                               "/run/mdtest/*")
+
+        # Running mdtest for different variants
+        for flag in mdtest_flags:
+            self.mdtest_cmd.flags.update(flag)
+            if self.mdtest_cmd.flags.value == ' ':
+                for api in mdtest_api:
+                    self.mdtest_cmd.api.update(api)
+                    if self.mdtest_cmd.api.value == 'DFS':
```

Comment:
Would recommend similar approach to map the variants as comma separated to avoid multiple if/else conditions.